gitlab-runner/src/uploader.rs

```diff
@@ -145,9 +156,15 @@ impl Uploader {
     pub(crate) async fn upload(self) -> JobResult {
         let (tx, rx) = oneshot::channel();
         self.tx.send(UploadRequest::Finish(tx)).await.unwrap();
-        let data = rx.await.unwrap().unwrap();
+        let file = AsyncFile::from_std(rx.await.unwrap().unwrap());
```

Comment:
```suggestion
        let file = rx.await().unwrap_or_else(| e | {
            warn!("Upload thread failed: {}");
            Err(()) 
         })?;
        let file = AsyncFile::from_std(file);
```
While there lets improve our error handling somewhat (untested)